operator/pkg/reconciliation/reconcile_reaper.go

```diff
@@ -0,0 +1,235 @@
+			{Name: "REAPER_CASS_CLUSTER_NAME", Value: dc.ClusterName},
+			{Name: "REAPER_CASS_CONTACT_POINTS", Value: fmt.Sprintf("[%s]", dc.GetSeedServiceName())},
+			{Name: "REAPER_AUTH_ENABLED", Value: "false"},
+			{Name: "REAPER_JMX_AUTH_USERNAME", Value: ""},
+			{Name: "REAPER_JMX_AUTH_PASSWORD", Value: ""},
+		},
+	}
+
+	return container
+}
+
+func getReaperImage(dc *api.CassandraDatacenter) string {
+	if len(dc.Spec.Reaper.Image) == 0 {
+		return ReaperDefaultImage
+	}
+	return dc.Spec.Reaper.Image
+}
+
+func getReaperPullPolicy(dc *api.CassandraDatacenter) corev1.PullPolicy {
+	if len(dc.Spec.Reaper.ImagePullPolicy) == 0 {
+		return ReaperDefaultPullPolicy
+	}
+	return dc.Spec.Reaper.ImagePullPolicy
+}
+
+func (rc *ReconciliationContext) CheckReaperSchemaInitialized() result.ReconcileResult {
+	// Using a job eventually get replaced with calls to the mgmt api once it has support for
+	// creating keyspaces and tables.
+
+	rc.ReqLogger.Info("reconcile_reaper::CheckReaperSchemaInitialized")
+
+	if !rc.Datacenter.Spec.Reaper.Enabled {
+		return result.Continue()
+	}
+
+	jobName := getReaperSchemaInitJobName(rc.Datacenter)
+	schemaJob := &v1batch.Job{}
+
+	err := rc.Client.Get(rc.Ctx, types.NamespacedName{Namespace: rc.Datacenter.Namespace, Name: jobName}, schemaJob)
+	if err != nil && errors.IsNotFound(err) {
+		// Create the job
+		schemaJob := buildInitReaperSchemaJob(rc.Datacenter)
+		rc.ReqLogger.Info("creating Reaper schema init job", ReaperSchemaInitJob, schemaJob.Name)
+		if err := setControllerReference(rc.Datacenter, schemaJob, rc.Scheme); err != nil {
+			rc.ReqLogger.Error(err, "failed to set owner reference", ReaperSchemaInitJob, schemaJob.Name)
+			return result.Error(err)
+		}
+		if err := rc.Client.Create(rc.Ctx, schemaJob); err != nil {
+			rc.ReqLogger.Error(err, "failed to create job", ReaperSchemaInitJob, schemaJob.Name)
+			return result.Error(err)
+		} else {
+			return result.RequeueSoon(2)
+		}
+	} else if err != nil {
+		return result.Error(err)
+	} else if jobFinished(schemaJob) {
+		return result.Continue()
+	} else {
+		return result.RequeueSoon(2)
+	}
+}
+
+func buildInitReaperSchemaJob(dc *api.CassandraDatacenter) *v1batch.Job {
+	return &v1batch.Job{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "Job",
+			APIVersion: "batch/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: dc.Namespace,
+			Name: getReaperSchemaInitJobName(dc),
+			Labels: dc.GetDatacenterLabels(),
+		},
+		Spec: v1batch.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					RestartPolicy: corev1.RestartPolicyOnFailure,
+					Containers: []corev1.Container{
+						{
+							Name:getReaperSchemaInitJobName(dc),
+							Image: ReaperSchemaInitJobImage,
+							ImagePullPolicy: corev1.PullAlways,
+							Env: []corev1.EnvVar{
+								{
+									Name: "KEYSPACE",
+									Value: ReaperKeyspace,
+								},
+								{
+									Name: "CONTACT_POINTS",
+									Value: dc.GetSeedServiceName(),
```

Comment:
So this is a DNS name that can be resolved to all the seeds in the datacenter. Does Reaper need every seed or just a single seed? I ask, because by default I'd assume it would resolve the DNS to just a single IP.

Reply:
Also, how does it handle seeds changing over time?

Reply:
>I ask, because by default I'd assume it would resolve the DNS to just a single IP

For a headless service I believe it gets mapped to all endpoints.

>how does it handle seeds changing over time?

I added a patch to Reaper a while back to handle C* node address changes. The behavior for this is controlled by the `REAPER_ENABLE_DYNAMIC_SEED_LIST` env var. Also note that Reaper does not care about seed nodes per se. It just needs an initial contact point, and from there it will discover other nodes via JMX.

Reply:
@johntrimble I may have misunderstood your initial point/question about seeds. Reaper only needs a single contact point; however, it is better to provide more than one. Reaper will go through the set of contact points until it establishes a successful connection.